api/client/credentials.go

```diff
@@ -201,16 +200,20 @@ type ProfileCreds struct {
 }
 
 // Dialer is used to dial a connection to Auth.
-func (c *ProfileCreds) Dialer(keepAliveInterval, dialTimeout time.Duration) (ContextDialer, error) {
+func (c *ProfileCreds) Dialer(cfg Config) (ContextDialer, error) {
+	if !cfg.InsecureAddressDiscovery {
+		return nil, trace.NotImplemented("cannot retrieve tunnel proxy address without an insecure connection")
+	}
+
 	sshConfig, err := c.SSHClientConfig()
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
-	dialer := NewTunnelDialer(*sshConfig, keepAliveInterval, dialTimeout)
+	dialer := NewTunnelDialer(*sshConfig, cfg.KeepAlivePeriod, cfg.DialTimeout)
 	return ContextDialerFunc(func(ctx context.Context, network, _ string) (conn net.Conn, err error) {
 		// Ping web proxy to retrieve tunnel proxy address.
-		pr, err := Find(ctx, c.profile.WebProxyAddr, false, nil)
+		pr, err := Find(ctx, c.profile.WebProxyAddr, true, nil)
```

Comment:
```suggestion
		pr, err := Find(ctx, c.profile.WebProxyAddr, cfg.InsecureAddressDiscovery, nil)
```

Reply:
If insecure is false, doesn't a valid `*x509.CertPool` need to be provided in place of nil?

Reply:
If `CertPool` is nil, `crypto/tls` will use the system pool